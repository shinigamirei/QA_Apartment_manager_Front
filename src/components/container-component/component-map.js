import { AdminHome, Admin1, Admin2, Admin3, Admin4, Admin5, Admin6,
         RecruiterHome, Recruiter1, Recruiter2, Recruiter3, Recruiter4, Recruiter5, Recruiter6 } from'./placeholder-components';
import Login from './login.component'
import React, { Component }  from 'react';
import ApartmentLanding from '../apartment_manager_landing.component';




let func = function(x){console.log(x)}


export const myConfig =
{
    "null": <Login/>,
    "recruiter": {
        "home": <RecruiterHome/>,
        "buttons": [
                        {'name': 'Home', 'content': <RecruiterHome/>}
                   ],
        "dropdowns":[
                        {   'name': 'Bursary',
                            'content':[
                                        {'name': 'Recruiter Bursary Link 1', 'content': <Recruiter1/>},
                                        {'name': 'Recruiter Bursary Link 2', 'content': <Recruiter2/>}
                                    ]
                        },
                        {  'name': 'Apartments',
                           'content':[
                                        {'name': 'Recruiter Apartment Link 1', 'content': <Recruiter3/>},
                                        {'name': 'Recruiter Apartment Link 2', 'content': <Recruiter4/>}
                                    ]
                        }
                    ],
        "side": [
                        {'name': 'Recruiter SideBar Link 1', 'content': <Recruiter5/>},
                        {'name': 'Recruiter SiderBar Link 2', 'content': <Recruiter6/>}
                   ],
    },
    "admin": {
        "home": <ApartmentLanding className="ApartmentLandingComponent"></ApartmentLanding>,
        "buttons": [
                        {'name': 'Home', 'content': <ApartmentLanding className="ApartmentLandingComponent"></ApartmentLanding>}
                   ],
        "dropdowns":[
                        {   'name': 'Demo Dropdown',
                            'content':[
                                        {'name': 'Admin placeholder 1', 'content':<AdminHome id="helloworld"/>},
                                        {'name': 'Admin placeholder 2', 'content':<Admin1/>},
                                        {'name': 'Admin placeholder 3', 'content': <Admin2/>},
                                        {'name': 'Admin placeholder 4', 'content': <Admin3/>},
                                        {'name': 'Admin placeholder 5', 'content': <Admin4/>}
                                      ]
                        }
                    ],
        "side": [
                        {'name': 'Admin Sidebar Link 1', 'content': <Admin5/>},
                        {'name': 'Admin Sidebar Link 2', 'content': <Admin6/>}
                   ],
    }
}