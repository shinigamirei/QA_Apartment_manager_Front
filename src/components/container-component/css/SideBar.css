/* Individual item */
.bm-item {
    display: inline-block;
  
    /* Our sidebar item styling */
    text-decoration: none;
    margin-bottom: 10px;
    color: #212529;
    transition: color 0.2s;
  }
  
  /* Change color on hover */
  .bm-item:hover {
    color: #5a6268;
  }
  
  /* The rest copied directly from react-burger-menu docs */
  
  /* Position and sizing of burger button */
  .bm-burger-button {
    opacity: 1.0;
    position: fixed;
    width: 25px;
    height: 25px;
    top: 70px;
    left: 20px;
  }

  .OpenMenu .bm-burger-button{
    width: 0px;
    height: 0px;
  }

  .bm-menu-wrap {
    z-index: 900 !important;
    position: fixed;
    bottom: 100;
    left: 1000;
    top: 60px !important;
    width: 12% !important;
  }
  
  /* Color/shape of burger icon bars */
  .bm-burger-bars {
    background: #373a47;
  }
  
  /* Position and sizing of clickable cross button */
  .bm-cross-button {
    height: 0px;
    width: 0px;
  }
  
  /* General sidebar styles */
  .bm-menu {
    background:#e6e6e7;
    padding: 2.5em 1.5em 0;
    font-size: 1.15em;
  }
  
  /* Morph shape necessary with bubble or elastic */
  .bm-morph-shape {
    fill: #373a47;
  }
  
  /* Wrapper for item list */
  .bm-item-list {
    color: #212529;
  }
  
  /* Styling of overlay */
  .bm-overlay {
    background: rgba(0, 0, 0, 0.3);
  }
