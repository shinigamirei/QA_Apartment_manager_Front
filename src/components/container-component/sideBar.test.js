import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow, mount, render, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MockAdapter from 'axios-mock-adapter';
import renderer from 'react-test-renderer';
import SideBar from './sideBar.component';
import { myConfig } from './component-map';

configure({ adapter: new Adapter() });

describe('Side Bar', () => {

    it('should render the side bar with admin links when role is admin', () => {
        const testRole = 'admin';

        const component = shallow(<SideBar role={testRole} links={myConfig}/>);

        expect(component).toMatchSnapshot();
    });

    it('should render the sidebar with recruiter links when role is recruiter', () => {
        const testRole = 'recruiter';

        const component = shallow(<SideBar role={testRole} links={myConfig}/>);

        expect(component).toMatchSnapshot();
    });

   
});