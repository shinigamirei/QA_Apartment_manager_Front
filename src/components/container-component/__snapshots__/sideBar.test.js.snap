// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Side Bar should render the side bar with admin links when role is admin 1`] = `
<div>
  <Menu
    bodyClassName=""
    burgerBarClassName=""
    burgerButtonClassName=""
    className=""
    crossButtonClassName=""
    crossClassName=""
    customBurgerIcon={
      <img
        src="Open.svg"
      />
    }
    customCrossIcon={
      <img
        src="Close.svg"
      />
    }
    disableAutoFocus={false}
    disableCloseOnEsc={false}
    htmlClassName="OpenMenu"
    id=""
    itemClassName=""
    itemListClassName=""
    menuClassName=""
    morphShapeClassName=""
    noOverlay={true}
    onStateChange={[Function]}
    outerContainerId="outer-container"
    overlayClassName=""
    pageWrapId="page-wrap"
    styles={Object {}}
    width={300}
  >
    <NavLink
      as={[Function]}
      disabled={false}
      onClick={[Function]}
    >
      Admin Sidebar Link 1
    </NavLink>
    <NavLink
      as={[Function]}
      disabled={false}
      onClick={[Function]}
    >
      Admin Sidebar Link 2
    </NavLink>
  </Menu>
</div>
`;

exports[`Side Bar should render the sidebar with recruiter links when role is recruiter 1`] = `
<div>
  <Menu
    bodyClassName=""
    burgerBarClassName=""
    burgerButtonClassName=""
    className=""
    crossButtonClassName=""
    crossClassName=""
    customBurgerIcon={
      <img
        src="Open.svg"
      />
    }
    customCrossIcon={
      <img
        src="Close.svg"
      />
    }
    disableAutoFocus={false}
    disableCloseOnEsc={false}
    htmlClassName="OpenMenu"
    id=""
    itemClassName=""
    itemListClassName=""
    menuClassName=""
    morphShapeClassName=""
    noOverlay={true}
    onStateChange={[Function]}
    outerContainerId="outer-container"
    overlayClassName=""
    pageWrapId="page-wrap"
    styles={Object {}}
    width={300}
  >
    <NavLink
      as={[Function]}
      disabled={false}
      onClick={[Function]}
    >
      Recruiter SideBar Link 1
    </NavLink>
    <NavLink
      as={[Function]}
      disabled={false}
      onClick={[Function]}
    >
      Recruiter SiderBar Link 2
    </NavLink>
  </Menu>
</div>
`;
